src/main/java/net/geforcemods/securitycraft/items/UniversalKeyChangerItem.java

```diff
@@ -69,4 +73,45 @@ else if(!(te instanceof DisguisableTileEntity) || (((BlockItem)((DisguisableBloc
 
 		return ActionResultType.FAIL;
 	}
+
+	@Override
+	public ActionResult<ItemStack> onItemRightClick(World world, PlayerEntity player, Hand hand) {
+		ItemStack keyChanger = player.getHeldItem(hand);
+
+		if (!world.isRemote) {
+			if (hand == Hand.MAIN_HAND && player.getHeldItemOffhand().getItem() == SCContent.BRIEFCASE.get()) {
+				ItemStack briefcase = player.getHeldItemOffhand();
+
+				if (isOwnedBy(player, briefcase)) {
+					if (briefcase.hasTag() && briefcase.getTag().contains("passcode")) {
+						if (!briefcase.hasTag())
+							briefcase.setTag(new CompoundNBT());
```

Comment:
This does not make sense. You check if the briefcase has a tag beforehand, so this can never be true.

Reply:
I do in fact check if the briefcase has a tag, and if it hasn't, the method returns true and the code continues. To not cause a NullPointerException, I check here again if it has a tag, and if no (which can in fact be the case), create one. I could also create the tag in `isOwnedBy`, but I didn't, because that method is only there to check for the owner, especially after your rewrite.

Reply:
In line 86 you say "if the briefcase has a tag and it's got a passcode". This means, the code in that if block will not be called if the briefcase has no tag. But the check I commented on is within that block, meaning it can never not have a tag.

Reply:
Oh. Oversaw that, sorry